course_flow/models.py

```diff
@@ -211,253 +252,161 @@ class Meta:
         verbose_name = "Node-Strategy Link"
         verbose_name_plural = "Node-Strategy Links"
 
-
-class Activity(models.Model):
-    title = models.CharField(max_length=30)
-    description = models.TextField(max_length=400)
-    author = models.ForeignKey(
-        User,
-        related_name="authored_activities",
-        on_delete=models.SET_NULL,
-        null=True,
-    )
+        
+class Workflow(models.Model):
+    objects = InheritanceManager()
+    
+    title = models.CharField(max_length=30, null=True, blank=True)
+    description = models.TextField(max_length=400, null=True, blank=True)
     created_on = models.DateTimeField(auto_now_add=True)
     last_modified = models.DateTimeField(auto_now=True)
 
     static = models.BooleanField(default=False)
 
-    students = models.ManyToManyField(
-        User, related_name="assigned_activities", blank=True
-    )
-
-    parent_activity = models.ForeignKey(
-        "Activity", on_delete=models.SET_NULL, null=True
+    parent_workflow = models.ForeignKey(
+        "Workflow", on_delete=models.SET_NULL, null=True
     )
     is_original = models.BooleanField(default=True)
 
     hash = models.UUIDField(default=uuid.uuid4, editable=False, unique=True)
 
     strategies = models.ManyToManyField(
-        Strategy, through="StrategyActivity", blank=True
+        Strategy, through="StrategyWorkflow", blank=True
     )
 
     columns = models.ManyToManyField(
-        Column, through="ColumnActivity", blank=True
+        Column, through="ColumnWorkflow", blank=True
     )
 
     outcomes = models.ManyToManyField(
-        Outcome, through="OutcomeActivity", blank=True
+        Outcome, through="OutcomeWorkflow", blank=True
     )
+    
+    SUBCLASSES = ["activity","course","program"]
+    
+    @property
+    def type(self):
+        for subclass in SUBCLASSES:
+            try:
+                return self[subclass].type
+            except Workflow[subclass].RelatedObjectDoesNotExist:
+                pass
+        return "workflow"
+    
+    def get_subclass(self):
```

Comment:
This seems to return the child object, rather than the child model, is that what you're going for?

Reply:
Yes, that's the intent